scrapy/downloadermiddlewares/redirect.py

```diff
@@ -71,6 +73,9 @@ def process_response(self, request, response, spider):
             return response
 
         location = safe_url_string(response.headers['location'])
+        if response.headers['location'].decode("latin1").startswith('/', 1):
```

Comment:
I'm not sure `latin1` is the best choice here, since `utf-8` is the [default encoding](https://github.com/scrapy/scrapy/blob/1.7.3/scrapy/http/headers.py#L10) for headers. Perhaps it would be better not to decode at all (i.e. perform the `startswith` check with a `bytes` object) or use [`scrapy.http.headers.Headers.to_unicode_dict`](https://github.com/scrapy/scrapy/blob/1.7.3/scrapy/http/headers.py#L82)

Also, I just learned about the `start` and `end` parameters of `str.startswith`. Would `startswith('//')` be clearer?

Reply:
I removed `decode("latin1")`, there is no need for it, as I can directly check with `startswith(b,'//')` on the bytes object.